sync-mixins.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# make a temporary dir to work in
+TMPDIR=$(mktemp --directory)
```

Comment:
Does `mktemp -d` works on linux ? 
As on Macos, it fails:
```
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix 
```

Reply:
Would you change it to `mktemp -d`

Reply:
Linux and MacOS compatibility is a nightmare in bash ... but well let's had `mktemp -d` if it works both ways